apiserver/pkg/versioneddataset/versioned_dataset.go

```diff
@@ -121,27 +121,23 @@ func VersionFiles(ctx context.Context, _ client.Client, input *generated.Version
 		return objectInfoList[i].LastModified.After(objectInfoList[j].LastModified)
 	})
 
-	existMap := make(map[string]struct{})
 	objMap := make(map[string][]string)
 	for _, obj := range objectInfoList {
 		objMap[obj.Key] = append(objMap[obj.Key], obj.VersionID)
 	}
-
 	result := make([]generated.PageNode, 0)
 	for _, obj := range objectInfoList {
-		if _, ok := existMap[obj.Key]; ok {
-			continue
-		}
-		if obj.IsDeleteMarker {
+		if !obj.IsLatest || obj.IsDeleteMarker {
 			continue
 		}
 		if keyword == "" || strings.Contains(obj.Key, keyword) {
-			existMap[obj.Key] = struct{}{}
 			lastModifiedTime := obj.LastModified
+			versionID := obj.VersionID
 			tf := generated.F{
-				Path:     strings.TrimPrefix(obj.Key, prefix),
-				Time:     &lastModifiedTime,
-				Versions: objMap[obj.Key],
+				Path:          strings.TrimPrefix(obj.Key, prefix),
+				Time:          &lastModifiedTime,
+				Versions:      objMap[obj.Key],
```

Comment:
Should we remove the version marked IsDeleteMarker?

Reply:
Not reqired.

Reply:
If the user selects a version marked with isDeleteMarker, an error will occur when selecting a file. So it should be filtered here.